qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -60,15 +71,27 @@ def __init__(self, parent=None, iface=None):
         self.hide_preview()
 
     def populate_resources(self, force_update=False):
-        response = get_all_resources(force_update=force_update)
-        # total = response.get("total")
-        # previous_url = response.get("previous")
-        # next_url = response.get("next")
-        resources = response.get("results", {})
+        if force_update or not self.resources:
+            response = get_all_resources(force_update=force_update)
+            # total = response.get("total")
+            # previous_url = response.get("previous")
+            # next_url = response.get("next")
+            self.resources = response.get("results", {})
+
+        geopackage_checked = self.checkBoxGeopackage.isChecked()
+        style_checked = self.checkBoxStyle.isChecked()
+        model_checked = self.checkBoxModel.isChecked()
+
+        filtered_resources = [
```

Comment:
Actually, I prefer to use `qsortfilterproxymodel` for filtering (and search also sorting in the future). Could you try to use `qsortfilterproxymodel`? It's more challenging to use of course :)

Reply:
I have implemented the `qsortfilterproxymodel` for filtering.